File: BengkelIn_SE/Models/Withdrawal.swift
//
//  Withdrawal.swift
//  BengkelIn_SE
//
//  Ported from MbengkelIn (Eugene's wallet feature).
//

import Foundation

struct Withdrawal: Codable, Identifiable {
    var id: String?
    var userId: String
    var amount: Double
    var bankName: String?
    var bankAccountNumber: String?
    var bankAccountName: String?
    var status: String
    var notes: String?
    var createdAt: Date?
    var updatedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case amount
        case bankName = "bank_name"
        case bankAccountNumber = "bank_account_number"
        case bankAccountName = "bank_account_name"
        case status
        case notes
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

File: admin/src/components/features/withdrawals/types/withdrawal.ts
import type { WithdrawalStatus } from "@/types/database"

export interface Withdrawal {
  id: string
  userId: string
  userName: string | null
  userEmail: string | null
  amount: number
  bankName: string | null
  bankAccountNumber: string | null
  bankAccountName: string | null
  status: WithdrawalStatus
  notes: string | null
  createdAt: string | null
  updatedAt: string | null
  userBalance: number | null
}

export interface WithdrawalActionResult {
  ok: boolean
  error?: string
}

export interface WithdrawalsTableProps {
  withdrawals: Withdrawal[]
}

export interface WithdrawalReviewDialogProps {
  withdrawal: Withdrawal
}
